### PR TITLE
Pull image in the wasi provider instead of hard coded

### LINKED ISSUE #96: Use non-hardcoded modules in wasi-provider
We are currently using a [hard coded WASM module](https://github.com/deislabs/krustlet/blob/master/crates/wascc-provider/src/lib.rs#L103) in the wasi-provider. This should be replaced with a module store that fetches the module if it's not there and returns the module from a cache.

### PR DESCRIPTION
**Pull request by rylev**
Fixes #96

### CODE REVIEW

File: examples/greet.yaml [hunk 1]
```diff
@@ -6,15 +6,15 @@ metadata:
     deislabs.io/wascc-action-key: MB4OLDIC3TCZ4Q4TGGOVAZC43VXFE2JQVRAXQMQFXUCREOOFEKOKZTY2
```
Comment (bacongobbler) [inline examples/greet.yaml]: this example appears to use the WasCC provider. should this be removed, or maybe a new example provided for the WASI provider instead?

File: crates/wasi-provider/src/lib.rs [hunk 2]
```diff
@@ -71,9 +73,20 @@ impl Provider for WasiProvider {
...
         let mut container_handles = HashMap::new();
         for container in pod.containers() {
+            let mut c = Client::default();
+
+            let image = container
+                .image
+                .clone()
+                .expect("Container must have an image");
+            let image = Reference::try_from(image).unwrap();
+            let store = FileModuleStore::new(&PathBuf::from("/home/rylevick"));
```
Comment (bacongobbler) [inline crates/wasi-provider/src/lib.rs]: should this reference a user's homedir rather than your home? :)

File: examples/greet.yaml (hunk 1 above)
Comment (thomastaylor312) [reply examples/greet.yaml]: I would prefer a wasi and wascc example

File: crates/wasi-provider/src/lib.rs (hunk 2 above)
Comment (technosophos) [reply crates/wasi-provider/src/lib.rs]: Good catch. Is this about the time when we define an env var for "KRUSTLET_HOME" or something

Comment (thomastaylor312) [inline crates/wasi-provider/src/lib.rs]: Should the store be a data member of the `WasiProvider` struct so we don't need to initialize it on every loop?

Comment (rylev) [reply crates/wasi-provider/src/lib.rs]: Yes. This should have been marked as WIP.

**Pull request merged**